### ISSUE TITLE
Problem with Yelp API and Server

### ISSUE THREAD

**Issue by ajax12233512** [caitlyn-griffing/ballard-street-cafe#14]
Aight so idk if anyones reading this. Lol i just got off work. Anyway, I really want to add the YELP api into our projects as it would be very cool imo. However, the only way I think we can do it is adding a server into our projects. Basically, all big APIs have a CORS request restriction that disallows any request from a client side application i.e. React. There are ways around it such as using allow-cors-anyware header but that would expose our API keys and i dont want that. 

What adding a server would mean is adding a server folder w/ a server.js inside of it. But you see, we cannot do that with the code we have now because the root of our git repository is our actual react app. (Meaning our root directory was configfured by Create-react-app).This means that our whole app is a front-end-app. Because we did that, webpack is secretly changing all the files when we start it with npm start so that it configures right. What we want to do is create a server folder that is outside the react-app. Meaning a folder that is outside the /ballard-street-cafe folder. It would have to be outside of our root and idk how to add files outside the root in vscode. I dont even want to try it. 

I think git submodules could be our best bet. It basically means having a git repo with another git repo inside of it. There are some subtle differences when working with git submodules but its not bad. I was thinking we could make another repo and make this the submodule of it. That way we can have a commit history of the server and app and also a commit history of just the app. 

This is all just speculation. Idk if its even worth it. I need to research more. 

TLDR: APIs are weird. Cant make proper request through react to Yelp. Might need to add server for api functionality.
